esphome/components/mcp4461/mcp4461.cpp

```diff
@@ -0,0 +1,662 @@
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return;
+  }
+  uint8_t wiper_idx = static_cast<uint8_t>(wiper);
+  ESP_LOGV(TAG, "Enabling terminal %c of wiper %" PRIu8 "", terminal, wiper_idx);
+  switch (terminal) {
+    case 'h':
+      this->reg_[wiper_idx].terminal_hw = true;
+      break;
+    case 'a':
+      this->reg_[wiper_idx].terminal_a = true;
+      break;
+    case 'b':
+      this->reg_[wiper_idx].terminal_b = true;
+      break;
+    case 'w':
+      this->reg_[wiper_idx].terminal_w = true;
+      break;
+    default:
+      ESP_LOGW(TAG, "Unknown terminal %c specified", terminal);
+      return;
+  }
+  this->reg_[wiper_idx].update_terminal = false;
+}
+
+void Mcp4461Component::disable_terminal_(Mcp4461WiperIdx wiper, char terminal) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return;
+  }
+  uint8_t wiper_idx = static_cast<uint8_t>(wiper);
+  ESP_LOGV(TAG, "Disabling terminal %c of wiper %" PRIu8 "", terminal, wiper_idx);
+  switch (terminal) {
+    case 'h':
+      this->reg_[wiper_idx].terminal_hw = false;
+      break;
+    case 'a':
+      this->reg_[wiper_idx].terminal_a = false;
+      break;
+    case 'b':
+      this->reg_[wiper_idx].terminal_b = false;
+      break;
+    case 'w':
+      this->reg_[wiper_idx].terminal_w = false;
+      break;
+    default:
+      ESP_LOGW(TAG, "Unknown terminal %c specified", terminal);
+      return;
+  }
+  this->reg_[wiper_idx].update_terminal = false;
+}
+
+uint16_t Mcp4461Component::get_eeprom_value(Mcp4461EepromLocation location) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return 0;
+  }
+  uint8_t reg = 0;
+  reg |= static_cast<uint8_t>(Mcp4461EepromLocation::MCP4461_EEPROM_1) + (static_cast<uint8_t>(location) * 0x10);
+  reg |= static_cast<uint8_t>(Mcp4461Commands::READ);
+  uint16_t buf;
+  if (!this->is_eeprom_ready_for_writing_(true)) {
+    return 0;
+  }
+  if (!this->read_byte_16(reg, &buf)) {
+    this->error_code_ = MCP4461_STATUS_I2C_ERROR;
+    this->status_set_warning();
+    ESP_LOGW(TAG, "Error fetching EEPRom location value");
+    return 0;
+  }
+  return buf;
+}
+
+bool Mcp4461Component::set_eeprom_value(Mcp4461EepromLocation location, uint16_t value) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return false;
+  }
+  uint8_t addr = 0;
+  if (value > 511) {
+    return false;
+  }
+  if (value > 256) {
+    addr = 1;
+  }
+  addr |= static_cast<uint8_t>(Mcp4461EepromLocation::MCP4461_EEPROM_1) + (static_cast<uint8_t>(location) * 0x10);
+  if (!(this->mcp4461_write_(addr, value, true))) {
+    this->error_code_ = MCP4461_STATUS_I2C_ERROR;
+    this->status_set_warning();
+    ESP_LOGW(TAG, "Error writing EEPRom value");
```

Comment:
Nitpicking here, but it's an acronym, should be all uppercase
```suggestion
    ESP_LOGW(TAG, "Error writing EEPROM value");
```